tools/clang/lib/SPIRV/TypeTranslator.cpp

```diff
@@ -438,13 +438,22 @@ uint32_t TypeTranslator::getElementSpirvBitwidth(QualType type) {
     case BuiltinType::LongLong:
     case BuiltinType::ULongLong:
       return 64;
-    // min16int (short), ushort, min12int, half, and min10float are treated as
-    // 16-bit if '-enable-16bit-types' option is enabled. They are treated as
+    // The HLSL 'half' keyword is translated to 'HalfFloat' enum internally,
+    // which can be 16 or 32 bits wide depending on command line options. The
```

Comment:
This one is for @ehsannas or @antiagainst.

Reply:
Could you share the sample code where you are seeing `half` mapped to `HalfFloat` with `-enable-16bit-types`?

Reply:
I just used a simple vertex shader with some `half` variables... nothing special.

Reply:
I'll double check my work, but I'd appreciate a second pair of eyes.

Reply:
Ok. Not sure if this is something specific to spirv path. At least on dxil path, I could see half mapped to `HalfTy` with `enable-16bit-types`,  As we could see that in the test `function_overload_selection_half_16bitflag.hlsl`, the overload selection works as expected for half. Currently I am working on accommodating @tex3d 's comments. Once done, I can take a look at spirv changes.

Reply:
Sorry it's a little bit confusing for me now. To clarify:

My previous thought is:

* Source code `half` keyword maps to AST `HalfFloat` builtin type. If we have `-enable-16bit-types`, `HalfFloat` should be 16-bit; otherwise it should be 32-bit.
* Source code `float16_t` keyword maps to AST `Half` builtin type. It cannot be used without `-enable-16bit-types` so it is always 16-bit.

According to @tex3d, the above is not true. Instead, we have

* Source code `half` maps to AST `HalfFloat` builtin type when `-enable-16bit-types` is given; it maps to AST `Half` builtin type when `-enable-16bit-types` is not given. 
* Source code `float16_t` keyword maps to AST `Half` builtin type. It cannot be used without `-enable-16bit-types` so it is always 16-bit.

Is the above understanding correct?

Reply:
Let's see if this clarifies:
- `Half` builtin type is always 16-bit.
- `HalfFloat` builtin type is just an alias for `Float` builtin type and is always 32-bit.
- if `-enable-16bit-types` is _true_, `half` keyword maps to `Half` builtin type.
- if `-enable-16bit-types` is _false_, `half` keyword maps to `HalfFloat` builtin type.

Reply:
Thanks, that's clear for me now.

Reply:
Great!